src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java

```diff
@@ -17,20 +17,23 @@ public class DatabaseFieldPreparer implements DefaultSubmissionFieldPreparer {
   public Map<String, SubmissionField> prepareSubmissionFields(Submission submission, PdfMap pdfMap) {
     Map<String, SubmissionField> databaseFields = new HashMap<>();
     Map<String, Object> dbFields = pdfMap.getDbFields();
-
-    dbFields.forEach((fieldName, value) -> {
-      switch (fieldName) {
-        case "submittedAt" ->
-            databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getSubmittedAt())));
-        case "submissionId" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getId().toString()));
-        case "createdAt" ->
-            databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getCreatedAt())));
-        case "updatedAt" ->
-            databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getUpdatedAt())));
-        case "flow" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getFlow()));
-        default -> log.error("Unable to map unknown database field: {}", fieldName);
-      }
-    });
+    
+    if (dbFields != null) {
+      dbFields.forEach((fieldName, value) -> {
+        switch (fieldName) {
+          case "submittedAt" ->
+              databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getSubmittedAt())));
+          case "submissionId" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getId().toString()));
+          case "createdAt" ->
+              databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getCreatedAt())));
+          case "updatedAt" ->
+              databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getUpdatedAt())));
+          case "flow" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getFlow()));
+          default -> log.error("Unable to map unknown database field: {}", fieldName);
```

Comment:
Do we need this? Do we not "fail early" if there are spurious fields in the YAML?

Reply:
For this case we do not. We do for the `flows-config` yaml but not for the `pdf-map` yaml. You make a good point though and we probably should be doing that. Maybe we make a ticket to do that outside of this ticket?